src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -75,7 +81,18 @@ IndexerConnector::IndexerConnector(
         }
     }
 
-    secureCommunication.sslCertificate(sslCertificate)
+    try
+    {
+        Keystore::get(INDEXER_COLUMN, USER_KEY, username);
+        Keystore::get(INDEXER_COLUMN, PASSWORD_KEY, password);
+    }
+    catch (const std::exception& e)
+    {
+        logError(IC_NAME, "%s", e.what());
+    }
```

Comment:
Include all function code inside of the first `try-catch` block:

```c++
    try
    {
        Keystore::get(INDEXER_COLUMN, USER_KEY, username);
        Keystore::get(INDEXER_COLUMN, PASSWORD_KEY, password);

        secureCommunication.basicAuth(username + ":" + password)
            .sslCertificate(sslCertificate)
            .sslKey(sslKey)
            .caRootCertificate(caRootCertificate);
.
.
.
                    }
                } while (!m_initialized && !m_cv.wait_for(lock, sleepTime, [&]() { return m_stopping.load(); }));
            });
    }
    catch (const std::exception& e)
    {
        logError(IC_NAME, "%s", e.what());
    }
}
```

Reply:
I think it is beyond the scope of this issue. I should only replace the code (deleted in the issue #21574):
```cpp
    if (config.contains("username") && config.contains("password"))
    {
        username = config.at("username").get_ref<const std::string&>();
        password = config.at("password").get_ref<const std::string&>();
    }
```
for:
```cpp
    try
    {
        Keystore::get(INDEXER_COLUMN, USER_KEY, username);
        Keystore::get(INDEXER_COLUMN, PASSWORD_KEY, password);
    }
    catch (const std::exception& e)
    {
        logError(IC_NAME, "%s", e.what());
    }
```

Reply:
An alternative solution is, if `Keystore::get()` would generate a potential exception, 
do not catch, to avoid continuing with the flow.